sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -128,32 +122,40 @@ namespace Azure { namespace Core {
     Context& operator=(const Context&) = default;
 
     /**
-     * @brief Create a context with expiration.
+     * @brief Create a context with a deadline.
      *
-     * @param cancelWhen A point in time after which a context expires.
+     * @param deadline A point in time after which a context expires.
      *
-     * @return A child context with expiration.
+     * @return A child context with deadline.
      */
-    Context WithDeadline(time_point cancelWhen) const
+    Context WithDeadline(DateTime const& deadline) const
     {
-      return Context{std::make_shared<ContextSharedState>(m_contextSharedState, cancelWhen)};
+      return Context{std::make_shared<ContextSharedState>(m_contextSharedState, deadline)};
     }
 
     /**
-     * @brief Create a context without an expiration, but with \p key and \p value associated with
-     * it.
+     * @brief Create a context without a deadline, but with \p key and \p value associated with it.
      *
      * @param key A key to associate with this context.
      * @param value A value to associate with this context.
      *
-     * @return A child context with no expiration and the \p key and \p value associated with it.
+     * @return A child context with no deadline and the \p key and \p value associated with it.
      */
     template <class T> Context WithValue(Key const& key, T&& value) const
     {
       return Context{std::make_shared<ContextSharedState>(
-          m_contextSharedState, (time_point::max)(), key, std::forward<T>(value))};
+          m_contextSharedState, (DateTime::max)(), key, std::forward<T>(value))};
     }
 
+    /**
+     * @brief Get a deadline time point associated with this context or the branch of contexts this
+     * context belongs to.
+     *
+     * @return A deadline associated with the context found; `Azure::DateTime::max()` value if a
```

Comment:
What does `Azure::DateTime::max()` mean and return? Is it public because of time_point max? https://en.cppreference.com/w/cpp/chrono/time_point/max

Reply:
I don't think it will matter in real life. So, setting it to now instead of min is fine with me.

Reply:
exactly